### PR TITLE
Reorder & tidy imports with rustfmt

### PR DESCRIPTION
**Pull request by ctz**
We can choose carefully when to merge this. The final commit is 100% mechanical, but it will cause significant conflicts in other PRs.

### CODE REVIEW

**Comment by ctz** [comment]
> 🟡 **Format** _Expected — Waiting for status to be reported_

We'd need to change the expected name of this job in the branch protection config.

**Comment by ctz** [comment]
Still not an opportune time to land this, but I reworked it to avoid stable `cargo fmt` seeing unstable options. Instead there's a `cargo fmt-unstable` alias which references the (separate) configuration file with them in.

File: .rustfmt.unstable.toml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+chain_width=40
```
Comment (djc) [inline .rustfmt.unstable.toml:1]: Tiny nit: spaces around the `=` signs?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -311,6 +311,29 @@ jobs:
...
 
+  format-unstable:
+    name: Format (unstable)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install rust nightly toolchain
```
Comment (djc) [inline .github/workflows/build.yml:322]: Maybe we should pin the nightly here?

**Review by djc — APPROVED** [review]

**Pull request merged**